--- gcc-4.6.3/gcc/libgcc2.c
+++ gcc-4.6.3/gcc/libgcc2.c
@@ -1355,7 +1355,6 @@
 __floatdixf (DWtype u)
 {
 #if W_TYPE_SIZE > XF_SIZE
-# error
 #endif
   XFtype d = (Wtype) (u >> W_TYPE_SIZE);
   d *= Wtype_MAXp1_F;
@@ -1369,7 +1368,6 @@
 __floatundixf (UDWtype u)
 {
 #if W_TYPE_SIZE > XF_SIZE
-# error
 #endif
   XFtype d = (UWtype) (u >> W_TYPE_SIZE);
   d *= Wtype_MAXp1_F;
@@ -1383,7 +1381,6 @@
 __floatditf (DWtype u)
 {
 #if W_TYPE_SIZE > TF_SIZE
-# error
 #endif
   TFtype d = (Wtype) (u >> W_TYPE_SIZE);
   d *= Wtype_MAXp1_F;
@@ -1397,7 +1394,6 @@
 __floatunditf (UDWtype u)
 {
 #if W_TYPE_SIZE > TF_SIZE
-# error
 #endif
   TFtype d = (UWtype) (u >> W_TYPE_SIZE);
   d *= Wtype_MAXp1_F;

