--- /home/esteban/helenos/coast-builds/ia32/build/libpng/libpng-1.6.10/pnglibconf.h	2014-03-13 09:29:02.954854511 -0300
+++ pnglibconf.h	2014-03-13 09:30:42.162849641 -0300
@@ -199,7 +199,7 @@
 #define PNG_TEXT_Z_DEFAULT_STRATEGY 0
 #define PNG_WEIGHT_SHIFT 8
 #define PNG_ZBUF_SIZE 8192
-#define PNG_ZLIB_VERNUM 0x1270
+#define PNG_ZLIB_VERNUM 0x1280
 #define PNG_Z_DEFAULT_COMPRESSION (-1)
 #define PNG_Z_DEFAULT_NOFILTER_STRATEGY 0
 #define PNG_Z_DEFAULT_STRATEGY 1
