=== modified file 'uspace/lib/posix/include/posix/stdlib.h'
--- uspace/lib/posix/include/posix/stdlib.h	2012-09-04 12:44:40 +0000
+++ uspace/lib/posix/include/posix/stdlib.h	2013-02-15 13:08:24 +0000
@@ -112,7 +112,8 @@
 extern void *posix_malloc(size_t size);
 extern void *posix_calloc(size_t nelem, size_t elsize);
 extern void *posix_realloc(void *ptr, size_t size);
-extern void posix_free(void *ptr);
+//extern void posix_free(void *ptr);
+extern void free(const void *ptr);
 
 /* Temporary Files */
 extern int posix_mkstemp(char *tmpl);
@@ -160,7 +161,7 @@
 	#define malloc posix_malloc
 	#define calloc posix_calloc
 	#define realloc posix_realloc
-	#define free posix_free
+	//#define free posix_free
 
 	#define mkstemp posix_mkstemp
 

=== modified file 'uspace/lib/posix/include/posix/strings.h'
--- uspace/lib/posix/include/posix/strings.h	2012-08-31 09:55:22 +0000
+++ uspace/lib/posix/include/posix/strings.h	2013-02-15 13:11:11 +0000
@@ -36,6 +36,8 @@
 #ifndef POSIX_STRINGS_H_
 #define POSIX_STRINGS_H_
 
+#include <libarch/types.h>
+
 /* Search Functions */
 #ifndef POSIX_STRING_H_
 extern int posix_ffs(int i);

=== modified file 'uspace/lib/posix/include/posix/unistd.h'
--- uspace/lib/posix/include/posix/unistd.h	2012-09-04 12:44:40 +0000
+++ uspace/lib/posix/include/posix/unistd.h	2013-02-15 13:18:21 +0000
@@ -76,7 +76,7 @@
 extern int posix_ftruncate(int fildes, posix_off_t length);
 extern int posix_rmdir(const char *path);
 extern int posix_unlink(const char *path);
-extern int posix_dup(int fildes);
+extern int dup(int fildes);
 extern int posix_dup2(int fildes, int fildes2);
 
 /* Standard Streams */
@@ -169,7 +169,7 @@
 	#define ftruncate posix_ftruncate
 	#define rmdir posix_rmdir
 	#define unlink posix_unlink
-	#define dup posix_dup
+	//#define dup posix_dup
 	#define dup2 posix_dup2
 
 	#define access posix_access

=== modified file 'uspace/lib/posix/source/stdlib.c'
--- uspace/lib/posix/source/stdlib.c	2012-08-31 09:55:22 +0000
+++ uspace/lib/posix/source/stdlib.c	2013-02-15 13:22:27 +0000
@@ -381,12 +381,12 @@
  *
  * @param ptr Memory chunk to be freed.
  */
-void posix_free(void *ptr)
+/*void posix_free(void *ptr)
 {
 	if (ptr) {
 		free(ptr);
 	}
-}
+}*/
 
 /**
  * Creates and opens an unique temporary file from template.

=== modified file 'uspace/lib/posix/source/unistd.c'
--- uspace/lib/posix/source/unistd.c	2012-08-31 09:55:22 +0000
+++ uspace/lib/posix/source/unistd.c	2013-02-15 13:18:16 +0000
@@ -272,7 +272,7 @@
  * @param fildes File descriptor to be duplicated.
  * @return On success, new file descriptor for the same file, otherwise -1.
  */
-int posix_dup(int fildes)
+int dup(int fildes)
 {
 	return posix_fcntl(fildes, F_DUPFD, 0);
 }

